### ISSUE TITLE
Unable to get into recent apps list with TalkBack

### ISSUE THREAD

**Issue by PatrykMis** [GrapheneOS/os-issue-tracker#1786]
Only in last alpha, previous versions worked well. I am getting these 2 stack traces interchangeably.

#### 1.

```
type: ANR
osVersion: google/flame/flame:13/TP1A.221005.002/2022120600:user/release-keys
package: android:33
process: system

activity: null
cause: ANR Input dispatching timed out (recents_animation_input_consumer (server) is not responding. Waited 5061ms for FocusEvent(hasFocus=false))
info: ANR in system
PID: 1508
Reason: Input dispatching timed out (recents_animation_input_consumer (server) is not responding. Waited 5061ms for FocusEvent(hasFocus=false))
ErrorId: 9ff32cf0-8927-432e-8f6a-25e1584014da
Frozen: false
Load: 2.95 / 2.95 / 1.94
----- Output from /proc/pressure/memory -----
some avg10=0.63 avg60=0.29 avg300=0.62 total=10182469
full avg10=0.48 avg60=0.23 avg300=0.43 total=7548952
----- End output from /proc/pressure/memory -----

CPU usage from 1869ms to -4444ms ago (2022-12-07 06:25:39.207 to 2022-12-07 06:25:45.520):
  36% 1508/system_server: 21% user + 14% kernel / faults: 21827 minor 379 major
  20% 674/surfaceflinger: 14% user + 5.5% kernel / faults: 2806 minor 8 major
  10% 143/kswapd0: 0% user + 10% kernel
  8.3% 1250/media.swcodec: 2.8% user + 5.5% kernel / faults: 9050 minor 22 major
  6.1% 2401/com.android.systemui: 3.1% user + 2.9% kernel / faults: 9374 minor 360 major
  5.5% 2452/app.talkbackfoss: 3.3% user + 2.2% kernel / faults: 5302 minor 498 major
  3.9% 676/android.hardware.graphics.composer@2.4-service-sm8150: 2.5% user + 1.4% kernel / faults: 2 minor
  2.5% 2391/com.android.phone: 1.4% user + 1.1% kernel / faults: 3054 minor 2325 major
  0% 3634/com.qualcomm.qti.services.secureui:sui_service: 0% user + 0% kernel / faults: 2954 minor 2352 major
  2.3% 272/kworker/u16:3: 0% user + 2.3% kernel
  0% 2323/.qtidataservices: 0% user + 0% kernel / faults: 3763 minor 2323 major
  2% 300/crtc_commit:125: 0% user + 2% kernel
  0% 2337/com.qti.phone: 0% user + 0% kernel / faults: 2182 minor 2209 major
  0% 3616/com.android.ims.rcsservice: 0% user + 0% kernel / faults: 2104 minor 2154 major
  1.8% 2291/com.android.networkstack.process: 0.9% user + 0.9% kernel / faults: 1986 minor 1299 major
  0% 2308/.dataservices: 0% user + 0% kernel / faults: 2023 minor 2115 major
  1.2% 277/kworker/u16:4: 0% user + 1.2% kernel
  0.9% 282/kworker/u16:8: 0% user + 0.9% kernel
  0.9% 1202/media.extractor: 0.3% user + 0.6% kernel / faults: 2765 minor 33 major
  0.1% 2353/org.codeaurora.ims: 0% user + 0% kernel / faults: 2153 minor 2097 major
  0.9% 2369/com.android.se: 0.3% user + 0.6% kernel / faults: 1962 minor 1842 major
  0.9% 10694/pl.tablica: 0.6% user + 0.3% kernel / faults: 424 minor
  0.6% 616/lmkd: 0.1% user + 0.4% kernel
  0.4% 6/kworker/u16:0: 0% user + 0.4% kernel
  0.4% 46/rcuop/4: 0% user + 0.4% kernel
  0.4% 76/kworker/u16:1: 0% user + 0.4% kernel
  0% 761/tombstoned: 0% user + 0% kernel / faults: 103 minor 12 major
  0% 1231/media.codec: 0% user + 0% kernel / faults: 1559 minor 39 major
  0.4% 3034/irq/32-90b6400.: 0% user + 0.4% kernel
  0.4% 3035/irq/33-90cd000.: 0% user + 0.4% kernel
  0.4% 4157/com.facebook.orca: 0.3% user + 0.1% kernel / faults: 6 minor 1 major
  0.3% 9/rcu_preempt: 0% user + 0.3% kernel
  0.3% 12/rcuop/0: 0% user + 0.3% kernel
  0.3% 30/rcuop/2: 0% user + 0.3% kernel
  0.3% 54/rcuop/5: 0% user + 0.3% kernel
  0.3% 615/logd: 0.1% user + 0.1% kernel / faults: 54 minor 1 major
  0.3% 673/android.hardware.power-service.pixel-libperfmgr: 0% user + 0.3% kernel / faults: 2 minor 2 major
  0.3% 877/statsd: 0.1% user + 0.1% kernel / faults: 112 minor 21 major
  0.3% 940/android.hardware.wifi@1.0-service: 0.3% user + 0% kernel / faults: 34 minor
  0.3% 3685/net.mullvad.mullvadvpn:mullvadvpn_daemon: 0% user + 0.3% kernel / faults: 545 minor 13 major
  0.3% 3942/com.facebook.katana: 0.1% user + 0.1% kernel / faults: 11 minor
  0.1% 10/rcu_sched: 0% user + 0.1% kernel
  0.1% 19/ksoftirqd/1: 0% user + 0.1% kernel
  0.1% 38/rcuop/3: 0.1% user + 0% kernel
  0.1% 62/rcuop/6: 0% user + 0.1% kernel
  0% 63/rcuos/6: 0% user + 0% kernel
  0.1% 70/rcuop/7: 0% user + 0.1% kernel
  0.1% 280/kworker/u16:7: 0% user + 0.1% kernel
  0.1% 301/crtc_event:125: 0% user + 0.1% kernel
  0.1% 350/kworker/3:2: 0% user + 0.1% kernel
  0% 536/kworker/2:1H: 0% user + 0% kernel
  0% 653/vold: 0% user + 0% kernel / faults: 66 minor 36 major
  0% 666/keystore2: 0% user + 0% kernel / faults: 138 minor 39 major
  0% 719/wlan_logging_th: 0% user + 0% kernel
  0.1% 878/netd: 0% user + 0.1% kernel / faults: 54 minor 16 major
  0.1% 879/zygote64: 0% user + 0.1% kernel / faults: 351 minor 1 major
  0.1% 968/msm_irqbalance: 0% user + 0.1% kernel
  0.1% 974/audioserver: 0% user + 0.1% kernel / faults: 21 minor 11 major
  0% 1109/drmserver: 0% user + 0% kernel / faults: 77 minor 69 major
  0.1% 1208/media.metrics: 0% user + 0.1% kernel / faults: 15 minor 6 major
  0.1% 1214/mediaserver: 0% user + 0.1% kernel / faults: 19 minor 8 major
  0.1% 2164/cds_ol_rx_threa: 0% user + 0.1% kernel
  0.1% 5409/com.whatsapp: 0% user + 0.1% kernel / faults: 21 minor 10 major
  0.1% 13154/kworker/0:3: 0% user + 0.1% kernel
 +0% 15964/com.android.nfc: 0% user + 0% kernel
17% TOTAL: 7.7% user + 8.8% kernel + 0.3% iowait + 0.5% irq + 0.2% softirq
CPU usage from 42ms to 603ms later (2022-12-07 06:25:41.119 to 2022-12-07 06:25:41.679):
  65% 1508/system_server: 34% user + 31% kernel / faults: 718 minor
    44% 15952/AnrConsumer: 18% user + 26% kernel
    13% 2111/binder:1508_7: 10% user + 2.6% kernel
    2.6% 1625/android.display: 2.6% user + 0% kernel
    2.6% 1626/android.anim: 2.6% user + 0% kernel
    2.6% 1627/android.anim.lf: 0% user + 2.6% kernel
    2.6% 2022/WifiHandlerThre: 0% user + 2.6% kernel
  29% 674/surfaceflinger: 20% user + 9% kernel / faults: 1 minor 1 major
    15% 674/surfaceflinger: 15% user + 0% kernel
    4.5% 1974/binder:674_4: 2.2% user + 2.2% kernel
    2.2% 715/binder:674_2: 0% user + 2.2% kernel
    2.2% 753/IdleTimer: 0% user + 2.2% kernel
    2.2% 760/surfaceflinger: 2.2% user + 0% kernel
  3.7% 143/kswapd0: 0% user + 3.7% kernel
  3.9% 272/kworker/u16:3: 0% user + 3.9% kernel
  1.8% 9/rcu_preempt: 0% user + 1.8% kernel
  1.8% 30/rcuop/2: 0% user + 1.8% kernel
  1.8% 54/rcuop/5: 0% user + 1.8% kernel
  1.9% 300/crtc_commit:125: 0% user + 1.9% kernel
  2.2% 676/android.hardware.graphics.composer@2.4-service-sm8150: 0% user + 2.2% kernel / faults: 1 minor
    4.5% 728/HwBinder:676_2: 2.2% user + 2.2% kernel
  2.2% 719/wlan_logging_th: 0% user + 2.2% kernel
  2.8% 2452/app.talkbackfoss: 2.8% user + 0% kernel / faults: 38 minor
    2.8% 2452/pp.talkbackfoss: 2.8% user + 0% kernel
  2.8% 3034/irq/32-90b6400.: 0% user + 2.8% kernel
  2.8% 3035/irq/33-90cd000.: 0% user + 2.8% kernel
17% TOTAL: 8.6% user + 7.9% kernel + 0.4% irq + 0.2% softirq

```

#### 2.

```
type: crash
osVersion: google/flame/flame:13/TP1A.221005.002/2022120600:user/release-keys
package: com.android.launcher3:33
process: com.android.launcher3
processUptime: 36826 + 457 ms

java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.accessibility.AccessibilityNodeInfo$AccessibilityAction com.android.launcher3.popup.SystemShortcut.createAccessibilityAction(android.content.Context)' on a null object reference
	at com.android.quickstep.views.TaskView.onInitializeAccessibilityNodeInfo(TaskView.java:1331)
	at android.view.View.createAccessibilityNodeInfoInternal(View.java:8677)
	at android.view.View.createAccessibilityNodeInfo(View.java:8662)
	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchDescendantsOfRealNode(AccessibilityInteractionController.java:1454)
	at android.view.AccessibilityInteractionController$AccessibilityNodePrefetcher.prefetchAccessibilityNodeInfos(AccessibilityInteractionController.java:1243)
	at android.view.AccessibilityInteractionController.findAccessibilityNodeInfoByAccessibilityIdUiThread(AccessibilityInteractionController.java:396)
	at android.view.AccessibilityInteractionController.-$$Nest$mfindAccessibilityNodeInfoByAccessibilityIdUiThread(Unknown Source:0)
	at android.view.AccessibilityInteractionController$PrivateHandler.handleMessage(AccessibilityInteractionController.java:1647)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7878)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)

```

**Comment by PatrykMis** [comment]
Update: crash occurres only when recents list is not empty.

To close apps, you can workarround: <https://stackoverflow.com/questions/33898029/adb-how-to-tap-close-from-recent-apps-to-completely-through-adb-one-liner-comma>

**Comment by thestinger** [comment]
The issue has been identified and will be resolved for the next release. The current release won't go beyond alpha.

**Comment by PatrykMis** [comment]
Great news, thanks for this! I've seen on Matrix this issue also affects Storage Scopes shortcut.

Btw. This was obviously not a TalkBack issue - I've tried compilation from Play Store too.

**Comment by thestinger** [comment]
It was a bug in my port of Storage Scopes shortcut in launcher to Android 13 QPR1. Next release resolves it.
